### PR TITLE
Document dependency verification update process

### PR DESCRIPTION
**Pull request by ljacomet**
Follow up on #18769

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -109,6 +109,14 @@ repositories {
 ```
 See [the documentation](userguide/kotlin_dsl.html#type-safe-accessors) for details.
 
+### Dependency verification improvements
+
+[Dependency verification](userguide/dependency_verification.html) is a feature that allows build to verify the checksums and signatures of the plugins and dependencies that are used by your project.
+
+With this release, the generation of the dependency verification file has been improved to produce stable output.
+This allows to leverage generation as an update strategy for the verification metadata when dependencies change in your project.
```
Comment (octylFractal) [inline subprojects/docs/src/docs/release/notes.md:117]:
```suggestion
This allows you to leverage generation as an update strategy for the verification metadata when dependencies change in your project.
```

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc [hunk 2]
```diff
@@ -142,11 +151,20 @@ Gradle won't verify either checksums or signatures of plugins which use their ow
...
 
-- the configuration file of the _current_ build is used for verification
-- so if the included build itself uses verification, its configuration is ignored in favor of the current one
-- which means that including a build works similarly to upgrading a dependency: it may require you to update your current verification metadata
+The verification file generated by Gradle has a strict ordering for all its content.
+It also uses the information from the existing state to limit changes to the strict minimum.
+
+This means that generation is actually a convenient tool for _updating_ a verification file:
+* Checksum entries generated by Gradle will have a clear `origin` that starts with "Generated by Gradle", which is a good indicator that an entry needs to be reviewed,
+* Entries added by hand will immediately be accounted for, and appear at their right location after writing the file,
```
Comment (octylFractal) [inline subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc:162]:
```suggestion
* Entries added by hand will immediately be accounted for, and appear at the right location after writing the file,
```

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc [hunk 3]
```diff
@@ -142,11 +151,20 @@ Gradle won't verify either checksums or signatures of plugins which use their ow
...
+* The header comments of the file will be preserved, that is comments before the root XML node.
```
Comment (octylFractal) [inline subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc:163]:
```suggestion
* The header comments of the file will be preserved, i.e. comments before the root XML node.
```

**Review by octylFractal — CHANGES_REQUESTED** [review]
Looks good overall, just a few grammar suggestions.

**Comment by ljacomet** [comment]
@bot-gradle test this

**Comment by ljacomet** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc [hunk 4]
```diff
@@ -142,11 +151,20 @@ Gradle won't verify either checksums or signatures of plugins which use their ow
...
+This allows to have license or instructions on which tasks and which parameters to use for generating that file.
+
+With the above benefits, it is really easy to account for new dependencies or dependency versions by simply generating the file again and reviewing the changes.
+This behaviour should make the feature much easier to use in practice.
```
[the remark this answers is not in the document]
Comment (tresat) [reply subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc]: I agree.  This: "This behaviour should make the feature much easier to use in practice." should be removed from this part of the documentaion.

**Review by octylFractal — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/docs/release/notes.md [hunk 5]
```diff
@@ -109,6 +109,14 @@ repositories {
 ```
 See [the documentation](userguide/kotlin_dsl.html#type-safe-accessors) for details.
 
+### Dependency verification improvements
+
+[Dependency verification](userguide/dependency_verification.html) is a feature that allows build to verify the checksums and signatures of the plugins and dependencies that are used by your project.
+
+With this release, the generation of the dependency verification file has been improved to produce stable output.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:116]: Could we clarify what specifically we mean by stable output?

File: subprojects/docs/src/docs/release/notes.md [hunk 6]
```diff
@@ -109,6 +109,14 @@ repositories {
...
+This allows you to leverage generation as an update strategy for the verification metadata when dependencies change in your project.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:117]: This is not entirely clear. Can we add some context on the "generation update strategy"?

**Review by pioterj — CHANGES_REQUESTED** [review]

**Pull request merged**